aspnetcore/fundamentals/routing.md

```diff
@@ -956,6 +966,14 @@ app.MapHealthChecks("/healthz");
 
 The routing tests have a [basic example](https://github.com/dotnet/AspNetCore/blob/main/src/Http/Routing/test/testassets/RoutingSandbox/Framework/FrameworkEndpointDataSource.cs#L17) of a non-updating data source.
 
+:::moniker-end
+:::moniker range=">= aspnetcore-7.0"
+
+**CONSIDER** implementing <xref:Microsoft.AspNetCore.Routing.EndpointDataSource.GetGroupedEndpoints%2A>. This gives complete control over running group conventions and the final metadata on the grouped endpoints.
```

Comment:
This is only relevant for library authors correct? For example, when is 

> This gives complete control over running group conventions and the final metadata on the grouped endpoints.

important?

Reply:
> This is only relevant for library authors correct?

Correct. This is in the `## Guidance for library authors` section.

> For example, when is [...] important?

What do you think about?

```suggestion
**CONSIDER** implementing <xref:Microsoft.AspNetCore.Routing.EndpointDataSource.GetGroupedEndpoints%2A>. This gives complete control over running group conventions and the final metadata on the grouped endpoints. For example, this allows custom `EndpointDataSource` implementations to run [endpoint filters](xref:fundamentals/minimal-apis/min-api-filters) added to groups.
```

The guidance we're currently giving for implementing `EndpointDataSource` at all is just above this.

> **CONSIDER** writing your own <xref:Microsoft.AspNetCore.Routing.EndpointDataSource>. `EndpointDataSource` is the low-level primitive for declaring and updating a collection of endpoints. `EndpointDataSource` is a powerful API used by controllers and Razor Pages.

So, these recommendations aren't intended to be super in-depth. I still haven't seen a custom `EndpointDataSource`, so I don't think it's worthwhile to spend much time expanding more on thius.